src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -132,11 +132,30 @@ public Test BuildFrom(FixtureMethod method, Test suite)
                     tests.Add(test); 
             }
 
-            return hasBuildersSpecified && method.Method.GetParameters().Length > 0 || tests.Count > 0
+            return hasBuildersSpecified && method.Method.GetParameters().Length > 0 || tests.Count > 0 ||
+                   HasMethodGotDataAttributeParameters(method) || 
+                   tests.Count > 0
```

Comment:
Two problems with checking here...

1) If there are data attributes, then there should already be a builder, one of the combining attributes, either specified or supplied by NUnit. Rather than adding a check, it would be better to figure out why there is no combining attribute for the data.

2. tests.Count > 0 is repeated twice.

Reply:
@CharliePoole On this line, for the tests that have zero cases, `hasBuildersSpecified` is false. `builders.Count` had been empty because we only look for builders from the method attributes, not parameter attributes.

> Rather than adding a check, it would be better to figure out why there is no combining attribute for the data.

I don't understand what this means. `hasBuildersSpecified` is false because there were no ITestBuilder attributes on the method. There is a `CombinatorialAttribute`, but that has never had any influence over what we do here unless it creates at least two tests.

Reply:
@jnm2 I'll wait till I'm actually in the code to reply... I've been doing too much today off the top of my head.

Reply:
It's just that I'm not following what you're saying. I debugged this branch and could be missing the obvious? Were you thinking we need a deeper refactor?